1Homework18.02.22/BWT/BWT/Program.cs

```diff
@@ -0,0 +1,313 @@
+                        {
+                            if (line[(arrayIndexes[j] + shift) % line.Length] <
+                                line[(arrayIndexes[start + (j - start) * 2 + 1] + shift) % line.Length])
+                            {
+                                Swap(ref arrayIndexes[j], ref arrayIndexes[start + (j - start) * 2 + 1]);
+                                j = start + (j - start) * 2 + 1;
+                            }
+                            else
+                            {
+                                j = i;
+                            }
+                        }
+                        else
+                        {
+                            if (line[(arrayIndexes[j] + shift) % line.Length] <
+                                line[(arrayIndexes[start + (j - start) * 2 + 2] + shift) % line.Length])
+                            {
+                                Swap(ref arrayIndexes[j], ref arrayIndexes[start + (j - start) * 2 + 2]);
+                                j = start + (j - start) * 2 + 2;
+                            }
+                            else
+                            {
+                                j = i;
+                            }
+                        }
+                    }
+                    else
+                    {
+                        if (line[(arrayIndexes[j] + shift) % line.Length] < line[(arrayIndexes[start + (j - start) * 2 + 1] + shift) % line.Length])
+                        {
+                            Swap(ref arrayIndexes[j], ref arrayIndexes[start + (j - start) * 2 + 1]);
+                        }
+
+                        j = i;
+                    }
+                }
+            }
+        }
+        
+        /// <summary>
+        /// Using Insertion sorting sorts a segment of the array,
+        /// which is a list of indexes of the characters of the transmitted word
+        /// </summary>
+        /// <param name="arrayIndexes">sortable array</param>
+        /// <param name="line">method where the line shifts are taken</param>
+        /// <param name="shift">the index of the character in the words, by which the sorting is currently performed</param>
+        /// <param name="start">start index</param>
+        /// <param name="end">end index</param>
+        private static void InsertionSort(int[] arrayIndexes, string line, int shift, int start, int end)
+        {
+            for (int i = start + 1; i <= end; ++i)
+            {
+                for (int j = i; j > start; --j)
+                {
+                    if (line[(arrayIndexes[j] + shift) % line.Length] <
+                        line[(arrayIndexes[j - 1] + shift) % line.Length])
+                    {
+                        Swap(ref arrayIndexes[j], ref arrayIndexes[j - 1]);
+                    }
+                }
+            }
+        }
+        /// <summary>
```

Comment:
```suggestion
        }
        
        /// <summary>
```
Тут и ниже, надо разделять методы пустыми строками